src/main.js

```diff
@@ -177,21 +162,12 @@ module.exports = {
         }
       }]
     }))
-
-    const initializeESLintWorker = () => {
-      this.worker = new Task(require.resolve('./worker.js'))
-    }
-    // Initialize the worker during an idle time
-    window.requestIdleCallback(initializeESLintWorker)
   },
 
   deactivate() {
-    if (this.worker !== null) {
-      this.worker.terminate()
-      this.worker = null
-    }
     idleCallbacks.forEach(callbackID => window.cancelIdleCallback(callbackID))
     idleCallbacks.clear()
+    helpers.killWorker()
```

Comment:
Note that this is _after_ the cancellation of the idle callbacks to prevent the idle callback that requests a worker start from firing off before it would be able to kill it.